src/analytics/publisher.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  DepartureGroupsPayload,
+  DepartureGroupsQuery,
+  DepartureRealtimeQuery,
+  DeparturesFromLocationPagingQuery,
+  FeatureLocation,
+  TripPatternsQuery
+} from '../service/types';
+import { PubSub, Topic } from '@google-cloud/pubsub';
+
+type Topics = {
+  departuresSearchGroupsTopic: string;
+  departuresSearchRealtimeTopic: string;
+  departuresSearchTopic: string;
+  getFeaturesTopic: string;
+  getTripPatternsTopic: string;
+};
+
+type Meta = {
+  environment: string;
+};
+
+export default (pubSub: PubSub, topics: Topics, meta: Meta) => {
+  const departuresSearchGroupsTopic = batchedTopic(
+    pubSub,
+    topics.departuresSearchGroupsTopic
+  );
+  const departuresSearchRealtimeTopic = batchedTopic(
+    pubSub,
+    topics.departuresSearchRealtimeTopic
+  );
+  const departuresSearchTopic = batchedTopic(
+    pubSub,
+    topics.departuresSearchTopic
+  );
+  const getFeaturesTopic = batchedTopic(pubSub, topics.getFeaturesTopic);
+  const getTripPatternsTopic = batchedTopic(
+    pubSub,
+    topics.getTripPatternsTopic
+  );
+
+  return {
+    async departuresSearchGroups(
+      payload: DepartureGroupsPayload,
+      query: DepartureGroupsQuery
+    ) {
+      await departuresSearchGroupsTopic.publish(serialize({ payload, query }), {
+        environment: meta.environment
+      });
```

Comment:
Nitpick, men kan sikkert lagre opts som eget objekt. Kan også unngå bruk av async/await her med å returnere resultatet av `publish` direkte. Da vil både `async` og `await` keyword bli overflødig

Reply:
Jeg ville helst unngå å lekke resultatet fra publish. Hva mener du med opts?

Reply:
Tenker at for å unngå å lekke resultatet så burde det try-catch-es eller ikke bruke async-await i det hele tatt (men da ville få warning med unhandled exception) for å ikke la feil lekke.

`opts` var ikke så farlig, så bare at samtlige publish kall brukte samme config. Om det er hensiktsmessig å tydliggjøre det med å bruke samme options-objekt eller om det ikke er det som er intensjonen her.